### ISSUE TITLE
LogZ in GAME 24

### ISSUE THREAD

**Issue by Xuekai-Zhu** [Yu-Fangxu/FoR#8]
Hi, Fangxu ! 
I have checked for all tasks for your repo. I find that the TB loss in GAME 24 uses the learnable LogZ, and the other tasks use the Eq.6 in paper.
GAME 24:
[Game24/util.py](https://github.com/Yu-Fangxu/FoR/blob/main/Game24/util.py)
```
logZ = torch.nn.Parameter(torch.tensor([args.logZ_init], dtype=torch.float))
def tb_loss(log_pf, log_r, logz):
    loss = (log_pf + logz - log_r) ** 2
    return loss.mean()
```
As for GSM8K, BlocksWorld:
```
def tb_loss(log_pf, log_r, logz, log_bf=None, logpartition=False):
    print("log_pf: ", log_pf)
    print("log_r: ", log_r)
    print("logz: ", logz)
    if logpartition:
        if log_bf != None:
            scores = log_pf - log_r - log_bf
            loss = (scores - scores.mean()) ** 2 
        else:
            scores = log_pf - log_r
            loss = (scores - scores.mean()) ** 2 
    else:
        if log_bf != None:
            loss = (log_pf + logz - log_r - log_bf) ** 2
        else:
            loss = (log_pf + logz - log_r) ** 2
    return loss.mean()
```

**Comment by 1609abhi** [comment]
hi are you able to run their code using 4bit = False on 80 Gb A100 on prontoqa task ?
